packages/fdc3-get-agent/src/strategies/getAgent.ts

```diff
@@ -20,87 +38,196 @@ export function getAgentPromise(): Promise<DesktopAgent> | null {
 }
 
 function initAgentPromise(options: GetAgentParams): Promise<DesktopAgent> {
-  const STRATEGIES = [new ElectronEventLoader(), new PostMessageLoader(), new TimeoutLoader()];
-  const promises = STRATEGIES.map(s => s.get(options));
+  Logger.log(`Initiating Desktop Agent discovery at ${new Date().toISOString()}`);
+  let strategies: Loader[];
 
-  return Promise.race(promises)
-    .then(da => {
-      // first, cancel the timeout etc.
-      STRATEGIES.forEach(s => s.cancel());
+  //if options doesn't contain an identityURL, use the actualUrl
+  if (!options.identityUrl) {
+    options.identityUrl = globalThis.window.location.href;
+  }
 
-      // either the timeout completes first with an error, or one of the other strategies completes with a DesktopAgent.
-      if (da) {
-        return da as DesktopAgent;
-      } else {
-        throw new Error(AgentError.AgentNotFound);
+  //Retrieve persisted connection data limit to a previous strategy if one exists
+  const persistedData = retrieveDesktopAgentDetails(options.identityUrl);
+  if (persistedData) {
+    switch (persistedData.agentType) {
+      case WebDesktopAgentType.Preload:
+        strategies = [new DesktopAgentPreloadLoader()];
+        break;
+      case WebDesktopAgentType.ProxyUrl:
+        //agentUrl will only be used by PostMessageLoader if not falsey
+        strategies = [new PostMessageLoader(persistedData.agentUrl)];
+        break;
+      case WebDesktopAgentType.ProxyParent:
+        strategies = [new PostMessageLoader()];
+        break;
+      case WebDesktopAgentType.Failover:
+        strategies = [];
+        break;
+      default:
+        strategies = [new DesktopAgentPreloadLoader(), new PostMessageLoader()];
```

Comment:
weirdly, this line isn't covered.  That implies we're not testing multiple strategies at the same time and the race.  We should do.

Reply:
I'll add a note to explain this - this line is only hit if the SessionsStorage has an unexpected/non-standard value in agentType. Using both strategies at once is the default path if there's no data in session storage which happens a couple of lines below.

We can pop a test in for an invalid agentType value to cover it.

Reply:
Test added